File: src/main/java/sorting/Bogosort.java
package sorting;

import java.util.Collections;
import java.util.List;
/*
 * bogosort o stupid sort semplicemente controlla se la lista e sorted
 * se non lo e shuffla tutti gli elementi finche non e sorted, con liste con tanti numeri
 * o numeri grandi puo metterci un sacco come metterci poco
 * 
 * notazione media O(n^2 * n!), caso ottimale O(n)
 */

public class Bogosort {
	public static boolean isSorted(List<Integer> arr) {
		for (int i = 1; i < arr.size(); i++) {
			if (arr.get(i) < arr.get(i - 1)) {
				return false;
			}
		}
		return true;
	}
	
	public static List<Integer> bogosort(List<Integer> arr){
		while(!isSorted(arr)) {
			Collections.shuffle(arr);
		}
		return arr;
	}
}

File: python/bogo_sort.py
"""
bogosort o stupid sort!!!
se arr non e sorted shuffla completamente 

notazione media O(n ^ 2 * !n)
"""
import random 
def isSorted(arr: list[int]) -> bool:
    for i in range(1, len(arr)):
        if arr[i] < arr[i - 1]:
            return False
    return True

def bogo_sort(arr: list[int]) -> None:
    while not isSorted(arr):
        random.shuffle(arr)
if __name__ == "__main__":
    l: list[int] = [1, 4, 3, 5, 2, 5, 7, 1]
    bogo_sort(l)
    print(l)
